pkg/app/gcp/instancemanager.go

```diff
@@ -204,6 +207,22 @@ func (m *InstanceManager) WaitOperation(zone string, user app.UserInfo, name str
 	return getter.Get()
 }
 
+func (m *InstanceManager) BuildHostReverseProxy(zone, host string) (*httputil.ReverseProxy, error) {
```

Comment:
My reasoning was:
- The IM knows how to connect to the instances it manages
- The IM knows whether the certificates should be verified or not (returning a URL doesn't provide this information)
- The current setup provides a direct network connection between the Cloud Orchestrator and the host machines, but that is not required. The connection could go through an agent/proxy, for example, a URL won't work for that either. The current setup actually uses an agent of sorts, the serverless VPC connector, it just happens to be at the network layer.

Returning a client that can send HTTP requests (GET and POST for now, the other ones could be added if needed) to the HO seems like the best abstraction to me: It hides as many details as possible from the underlying architecture and only lets users of the interface do HTTP. Another example that a URL won't work for is if you deploy the entire system is a single server and decide to use UNIX sockets to communicate between CO and HO.

That being said, returning a reverse proxy does feel like a **hack** and I added it because there is no convenient way to create one from an HTTP client. I think it could be cleaner if we move the `build-reverse-proxy` method to the client interface, WDYT?

Reply:
In case it wasn't evident: I intentionally removed the `GetURL` method from the interface.

Reply:
We can continue the discussion here: 
https://github.com/google/cloud-android-orchestration/pull/141/files/a4c8a69e00f533f94885859fb83551a9f5f29225#r1171699696.

From there I mention that we should only keep `GetHostAddr` in the IM interface get rid of the `GetHostURL`, `BuildHostReverseProxy` and `GetHostHTTPClient` from here and put the logic behind these methods in a different component cloud provider agnostic.

I don't argue against `GetHostURL` vs `GetHostHTTPClient`, now that we have a more complex setup, I think these two could be merged into just one, but in a different component.

Reply:
That link doesn't work for me.